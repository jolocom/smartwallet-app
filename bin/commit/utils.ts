import childProcess from 'child_process'

export const promisify =
  <T>(fn: (args: T) => void) =>
  (args: T): Promise<void> =>
    new Promise((res, rej) => {
      try {
        fn(args)
        res()
      } catch (e) {
        rej(e)
      }
    })

export const listStagedFiles = () =>
  childProcess.execSync(`git diff --cached --name-only --diff-filter=ACMR`)

export const stageModifiedFiles = (stagedFiles: string) =>
  childProcess.execSync(`echo "${stagedFiles}" | xargs git add`)

export const abortScript = (msg: string) => {
  console.log('')
  console.log('\x1b[0;33m%s\x1b[0m', msg)
  console.log('\x1b[0;31m%s\x1b[0m', 'Aborting')
  process.exit(1)
}

export const logStep = (msg: string) => {
  console.log('\x1b[0;34m%s\x1b[0m', `${msg}`)
}

export const spawnProcess = (
  onClose: (
    code: number | null,
    dataOutput: string,
    errorOutput: string,
  ) => void,
  cmd: string,
  args?: readonly string[],
  options?: childProcess.SpawnOptionsWithoutStdio,
) => {
  const spawnedProcess = childProcess.spawn(cmd, args, options)

  let dataOutput = ''
  let errorOutput = ''

  spawnedProcess.stdout.on('data', (data) => {
    dataOutput += data.toString() + '\n'
  })
  spawnedProcess.stderr.on('data', (error: Buffer) => {
    errorOutput += error.toString() + '\n'
  })

  spawnedProcess.stderr.pipe(process.stderr)
  spawnedProcess.stdout.pipe(process.stdout)

  spawnedProcess.on('close', (code) => onClose(code, dataOutput, errorOutput))

  return spawnedProcess
}
