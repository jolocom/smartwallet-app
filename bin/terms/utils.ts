import fetch from 'node-fetch';
import fs from 'fs';
import shell from 'shelljs'

const CURRENT_PATH = process.cwd();
const SECRETS_REPO = 'dev@hetz1.jolocom.io:poeditor-key'
const BASE_URL = 'https://api.poeditor.com/v2';
const API_KEY_LOCATION = `${CURRENT_PATH}/bin/poeditor-key/POEDITOR_TOKEN.txt`
const PROJECT_ID_LOCATION = `${CURRENT_PATH}/bin/poeditor-key/POEDITOR_SW2_PROJECT_ID.txt`

export enum Languages {
  en = 'en',
  de = 'de',
}

interface IResult  {
  [key: string]: {
    [key: string]: number
  }
}

export interface IResponse<T = IResult> {
  response: {
    status: string,
    code: string,
    message: string,
  },
  result: T
}

export const cloneSecrets = () => {
  return new Promise((res, rej) => {
    const keyExists = fs.existsSync(API_KEY_LOCATION);
    const idExists = fs.existsSync(PROJECT_ID_LOCATION);
    if(!keyExists || !idExists) {
      shell.cd(`${CURRENT_PATH}/bin`)
      shell.rm('-rf', `poeditor-key`)
      shell.exec(
        `git clone ${SECRETS_REPO} -b master`,
        {},
        (code, stdout, stderr) => {
          if (code != 0) return rej(new Error(stderr))
          return res(stdout)
        },
      )
    } else {
      res('Secrets are already present');
    }
  })
}

const readTextFile = (location: string) => {
  return new Promise<string>((res, rej) => {
    fs.readFile(location, 'utf8', (err, data) => {
      if (err) rej(err)
      else res(data.toString().trim())
    })
  })
}

const getPoeditorSecrets = async () => {
    const apiKey = await readTextFile(API_KEY_LOCATION)
    const projectId = await readTextFile(PROJECT_ID_LOCATION)
    return { apiKey, projectId }
}


export const sendPostRequest = async <T = IResponse, >(path: string, params: Record<string, string>) => {
  try {
    const { apiKey, projectId } = await getPoeditorSecrets();
    const requestParams = new URLSearchParams({
      api_token: apiKey,
      id: projectId,
      ...params
    })
    const res = await fetch(`${BASE_URL}${path}`, {
      method: 'POST',
      body: requestParams.toString(),
      headers: {
        'Content-Type': 'application/x-www-form-urlencoded'
      }
    })
    return await res.json() as T;
  } catch(err) {
    console.error(err);
  }
}