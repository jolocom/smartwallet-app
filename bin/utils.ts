import fs from 'fs'
import { stringify as stringifyEnv, parse as parseEnv, Data } from 'envfile'
import shell from 'shelljs'

export const cloneRepo = async (remote: string, repoName: string) => {
  return new Promise<string>((res, rej) => {
    const dir = shell.tempdir()
    console.log(`${dir}/${repoName}`)
    shell.rm('-rf', `${dir}/${repoName}`)
    shell.exec(
      `git clone ${remote}:${repoName} -b master ${dir}/${repoName}`,
      {},
      (code, stdout, stderr) => {
        if (code != 0) return rej(new Error(stderr))
        return res(`${dir}/${repoName}`)
      },
    )
  })
}

export const catFile = async (path: string) => {
  return new Promise((res, rej) => {
    const dsn = shell.cat(path)
    if (dsn.stderr) rej(dsn.stderr)
    res(dsn.stdout)
  })
}

export const writeEnv = async (path: string, data: object) => {
  const env = stringifyEnv(data)

  return new Promise((res, rej) => {
    fs.writeFile(path, env, (err) => {
      if (err) rej(err)
      else res(undefined)
    })
  })
}

export const readEnv = async (path: string) => {
  return new Promise<Data>((res, rej) => {
    fs.readFile(path, 'utf8', (err, data) => {
      if (err) {
        console.log(
          '\x1b[4;31m%s\x1b[0m',
          'Error! Could not find the environmental config (.env).',
        )
      } else res(parseEnv(data))
    })
  })
}
