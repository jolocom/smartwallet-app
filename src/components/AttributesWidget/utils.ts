import { AttrsState } from '~/modules/attributes/types'
import { SelectableAttrI } from './index'

export const getMappedAttrubited = (attrs: AttrsState<string>) =>
  Object.keys(attrs).reduce((acc, val) => {
    acc[val] = attrs[val].map((attr) => ({
      val: attr,
      isSelected: false,
    }))
    return acc
  }, {} as AttrsState<SelectableAttrI>)
