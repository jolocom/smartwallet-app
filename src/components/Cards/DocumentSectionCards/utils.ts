export const splitFields = <T>(fields: Array<T>) => {
  const halfway = Math.ceil(fields.length / 2)
  const last = fields.splice(halfway, fields.length)
  const first = fields.splice(0, halfway)

  return [first, last]
}
