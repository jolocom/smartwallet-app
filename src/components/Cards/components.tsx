import React, { ReactNode, useEffect } from 'react'
import {
  Image,
  ImageBackground,
  StyleProp,
  StyleSheet,
  TextStyle,
  TouchableOpacity,
  View,
  ViewStyle,
} from 'react-native'
import LinearGradient from 'react-native-linear-gradient'
import { PurpleTickSuccess } from '~/assets/svg'
import { DocumentProperty } from '~/hooks/documents/types'
import useImagePrefetch from '~/hooks/useImagePrefetch'
import useTranslation from '~/hooks/useTranslation'
import { TextLayoutEvent } from '~/types/props'
import { Colors } from '~/utils/colors'
import { Fonts } from '~/utils/fonts'
import JoloText from '../JoloText'
import { DOCUMENT_HEADER_HEIGHT } from './consts'
import { useCredentialNameScale, usePruneFields } from './hooks'
import { ScaledText, ScaledView } from './ScaledCard'
import { splitIntoRows } from './utils'

export const FieldsCalculator: React.FC<{
  cbFieldsVisibility: (child: ReactNode, idx: number) => ReactNode
}> = ({ children, cbFieldsVisibility }) =>
  React.Children.map(children, cbFieldsVisibility) as React.ReactElement<
    unknown,
    string
  > | null

export const CardMoreBtn: React.FC<{
  onPress?: () => void
}> = ({ onPress }) => (
  <ScaledView
    scaleStyle={styles.dotsContainerScaled}
    style={styles.dotsContainer}
  >
    <TouchableOpacity onPress={onPress} style={styles.dotsBtn}>
      <ScaledView scaleStyle={styles.scaledDots}>
        {[...Array(3).keys()].map((c) => (
          <ScaledView key={c} scaleStyle={styles.dot} />
        ))}
      </ScaledView>
    </TouchableOpacity>
  </ScaledView>
)

export const DocumentFooter: React.FC<{
  leftIcons?: string[]
  renderRightIcon?: () => JSX.Element
  expired?: boolean
  style?: StyleProp<ViewStyle>
}> = ({ renderRightIcon, leftIcons, style = {}, expired = false }) => {
  const { t } = useTranslation()

  return (
    <ScaledView
      style={[styles.footerContainer, style]}
      scaleStyle={styles.footerContainerScaled}
    >
      <View style={styles.footerBorder} />
      <View style={styles.footerContent}>
        {expired ? (
          <View style={styles.footerExpiredContainer}>
            <JoloText
              customStyles={{ fontFamily: Fonts.Medium }}
              color={Colors.mainBlack}
            >
              {t('DocumentCard.expired')}
            </JoloText>
          </View>
        ) : (
          <>
            <View style={styles.footerIconsContainer}>
              {leftIcons &&
                leftIcons.map((icon, i) => (
                  <ScaledView
                    key={i}
                    scaleStyle={{ width: 40, height: 30 }}
                    style={{ marginRight: 10 }}
                  >
                    <Image
                      source={{ uri: icon }}
                      resizeMode="contain"
                      style={{
                        width: '100%',
                        height: '100%',
                        borderRadius: 4.2,
                      }}
                    />
                  </ScaledView>
                ))}
            </View>
            <View>{renderRightIcon && renderRightIcon()}</View>
          </>
        )}
      </View>
    </ScaledView>
  )
}

export const DocumentPhoto: React.FC<{
  photo: string
}> = ({ photo }) => (
  <ScaledView
    scaleStyle={[styles.photoContainerScaled]}
    style={styles.photoContainer}
  >
    <Image resizeMode="cover" style={styles.photo} source={{ uri: photo }} />
  </ScaledView>
)

export const DocumentHeader: React.FC<{
  name: string
  icon?: string
  selected?: boolean
  backgroundImage?: string
  backgroundColor?: string
}> = ({ name, icon, selected, backgroundColor, backgroundImage }) => {
  const { handleCredentialNameTextLayout } = useCredentialNameScale()

  const prefetchedIcon = useImagePrefetch(icon)

  const renderBackground = (children: () => React.ReactChild) => {
    if (backgroundImage) {
      return (
        <DocumentBackgroundImage image={backgroundImage}>
          {children()}
        </DocumentBackgroundImage>
      )
    } else if (backgroundColor) {
      return (
        <DocumentBackgroundColor color={backgroundColor}>
          {children()}
        </DocumentBackgroundColor>
      )
    } else {
      return children()
    }
  }

  return (
    <View>
      {renderBackground(() => (
        <ScaledView
          scaleStyle={{
            height: DOCUMENT_HEADER_HEIGHT,
            padding: 16,
          }}
          style={styles.headerContainer}
        >
          {prefetchedIcon && (
            <ScaledView
              scaleStyle={{
                width: 32,
                height: 32,
                marginRight: 10,
              }}
            >
              <Image
                resizeMode="cover"
                style={[styles.photo, { borderRadius: 4.2 }]}
                source={{ uri: prefetchedIcon }}
              />
            </ScaledView>
          )}
          <ScaledView
            scaleStyle={styles.credentialName}
            style={{ flex: 1, flexDirection: 'row' }}
          >
            <ScaledText
              // @ts-expect-error
              onTextLayout={handleCredentialNameTextLayout}
              numberOfLines={1}
              scaleStyle={styles.credentialName}
              style={[styles.mediumText]}
            >
              {name}
            </ScaledText>
            <View style={{ flex: 1 }} />
          </ScaledView>
          {typeof selected !== 'undefined' && (
            <SelectedToggle selected={selected} />
          )}
        </ScaledView>
      ))}
    </View>
  )
}

export const DocumentHolderName: React.FC<{
  name: string
  onLayout?: (e: TextLayoutEvent) => void
  cropName?: boolean
  numberOfLines?: number
}> = ({ name, onLayout, cropName = false, numberOfLines = 2 }) => {
  return (
    <ScaledView
      scaleStyle={{
        paddingLeft: 24,
        marginRight: cropName ? 116 : 0,
        marginBottom: 8,
      }}
    >
      <ScaledText
        // @ts-expect-error
        onTextLayout={onLayout}
        numberOfLines={numberOfLines}
        style={styles.mediumText}
        scaleStyle={styles.holderName}
      >
        {name}
      </ScaledText>
    </ScaledView>
  )
}

export const DocumentFields: React.FC<{
  // NOTE: fields to be (potentially) displayed on the card
  fields: DocumentProperty[]
  // NOTE: (scaled) styles for the field label
  labelScaledStyle: TextStyle
  // NOTE: (scaled) styles for the field value
  valueScaledStyle: TextStyle
  // NOTE: max number of lines to be displayed per field
  maxLines: number
  // NOTE: max number of rows that can be displayed
  maxRows: number
  // NOTE: distance between lines
  rowDistance?: number
  // NOTE: number of characters to allow a field value to be displayed in a single line
  fieldCharacterLimit?: number
  // NOTE: number of columns to render the fields in
  nrOfColumns?: number
  // NOTE: allow fields taking up available space if possible
  allowOverflowingFields?: boolean
  // NOTE: called after calculating the fields that will be displayed
  onFinishCalculation?: (displayedFields: DocumentProperty[]) => void
}> = ({
  fields,
  maxLines,
  maxRows,
  rowDistance = 0,
  fieldCharacterLimit = 14,
  valueScaledStyle,
  labelScaledStyle,
  nrOfColumns = 2,
  onFinishCalculation,
  allowOverflowingFields = true,
}) => {
  const maxFields = maxRows * 2
  const { displayedFields, handleFieldValuesVisibility } = usePruneFields(
    fields,
    maxFields,
    maxLines,
  )

  let rows = splitIntoRows(displayedFields, fieldCharacterLimit, nrOfColumns)
  // NOTE: since when splitting we may get more rows than @maxRows due to the value overflowing,
  // we have to cut it to the max nr of rows.
  rows = rows.splice(0, maxRows)

  useEffect(() => {
    const fields = rows.reduce<DocumentProperty[]>((acc, row) => {
      acc = [...acc, ...row]

      return acc
    }, [])

    onFinishCalculation && onFinishCalculation(fields)
  }, [rows])

  const renderField = (field: DocumentProperty) => {
    return (
      <ScaledView
        key={field.key}
        style={{ flex: 1 }}
        scaleStyle={{ paddingRight: 12 }}
      >
        <ScaledText
          numberOfLines={1}
          style={[
            styles.regularText,
            {
              width: '100%',
            },
          ]}
          scaleStyle={[styles.fieldLabel, labelScaledStyle]}
        >
          {field?.label?.trim()}:
        </ScaledText>
        <ScaledText
          numberOfLines={1}
          scaleStyle={[styles.fieldText, valueScaledStyle]}
          style={[
            styles.mediumText,
            {
              width: '100%',
            },
          ]}
        >
          {field?.value}
        </ScaledText>
      </ScaledView>
    )
  }

  return (
    <ScaledView
      scaleStyle={{
        paddingLeft: 24,
      }}
      style={{
        flex: 1,
        justifyContent: 'center',
      }}
    >
      <FieldsCalculator cbFieldsVisibility={handleFieldValuesVisibility}>
        <View
          style={{
            flex: 1,
            alignItems: 'flex-start',
          }}
        >
          {rows.map((row, idx) => (
            <ScaledView
              key={idx}
              style={{
                flexDirection: 'row',
                alignItems: 'flex-start',
              }}
              scaleStyle={{
                marginTop: idx === 0 ? 0 : rowDistance,
              }}
            >
              {row.map(renderField)}
              {!allowOverflowingFields && row.length % 2 === 1 && (
                <View style={{ flex: 1 }} />
              )}
            </ScaledView>
          ))}
        </View>
      </FieldsCalculator>
    </ScaledView>
  )
}

const BackgroundOpacity: React.FC = ({ children }) => (
  <LinearGradient
    colors={[Colors.white100, Colors.white00]}
    style={{ flex: 1 }}
  >
    {children}
  </LinearGradient>
)

export const DocumentBackgroundImage: React.FC<{ image: string }> = ({
  image,
  children,
}) => (
  <ScaledView
    scaleStyle={{ height: 112 + DOCUMENT_HEADER_HEIGHT }}
    style={{ width: '100%' }}
  >
    <ImageBackground
      style={{ width: '100%', height: '100%' }}
      source={{ uri: image }}
    >
      <View style={{ zIndex: 99 }}>{children}</View>
      <BackgroundOpacity />
    </ImageBackground>
  </ScaledView>
)

export const DocumentBackgroundColor: React.FC<{ color: string }> = ({
  color,
  children,
}) => (
  <ScaledView
    scaleStyle={{
      height: 84 + DOCUMENT_HEADER_HEIGHT,
    }}
    style={{
      width: '100%',
      backgroundColor: color,
    }}
  >
    <BackgroundOpacity>{children}</BackgroundOpacity>
  </ScaledView>
)

export const SelectedToggle: React.FC<{ selected: boolean }> = ({
  selected,
}) => {
  return (
    <ScaledView scaleStyle={styles.selectIndicator}>
      {selected ? (
        <PurpleTickSuccess />
      ) : (
        <ScaledView
          scaleStyle={styles.notSelectedScale}
          style={styles.notSelected}
        />
      )}
    </ScaledView>
  )
}

const styles = StyleSheet.create({
  dotsContainerScaled: {
    top: 18,
    right: 18,
    paddingVertical: 3,
    width: 20,
    height: 30,
  },
  dotsContainer: {
    justifyContent: 'center',
    zIndex: 99,
    position: 'absolute',
    height: '100%',
    borderRadius: 8,
  },
  scaledDots: {
    flex: 1,
    width: '100%',
    height: '100%',
    alignItems: 'center',
    paddingVertical: 2,
    justifyContent: 'space-between',
    flexDirection: 'column',
  },
  dotsBtn: {
    flex: 1,
  },
  dot: {
    width: 4,
    height: 4,
    borderRadius: 2,
    marginHorizontal: 2,
    backgroundColor: Colors.black,
  },
  credentialName: {
    fontSize: 22,
    lineHeight: 26,
    fontWeight: '500',
    paddingHorizontal: 2,
  },
  holderName: {
    fontSize: 24,
    lineHeight: 28,
  },
  photoContainer: {
    position: 'absolute',
    overflow: 'hidden',
    zIndex: 10,
  },
  photoContainerScaled: {
    right: 14,
    width: 82,
    height: 82,
    borderRadius: 41,
  },
  headerContainer: {
    width: '100%',
    alignItems: 'center',
    flexDirection: 'row',
  },
  photo: {
    width: '100%',
    height: '100%',
  },
  footerContainerScaled: {
    height: 60,
    paddingHorizontal: 20,
  },
  footerContainer: {
    width: '100%',
    zIndex: 9,
  },
  footerBorder: {
    width: '100%',
    borderTopWidth: 1,
    borderTopColor: '#D8D8D8',
  },
  footerContent: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    height: '100%',
  },
  footerExpiredContainer: {
    paddingHorizontal: 10,
    paddingVertical: 5,
    backgroundColor: Colors.yellow,
    borderRadius: 9,
  },
  footerIconsContainer: {
    flexDirection: 'row',
    justifyContent: 'flex-start',
  },
  regularText: {
    fontFamily: Fonts.Regular,
    fontSize: 22,
    color: Colors.black,
  },
  mediumText: {
    fontFamily: Fonts.Medium,
    color: Colors.black,
  },
  fieldLabel: {
    fontSize: 16,
    lineHeight: 16,
    color: Colors.slateGray,
  },
  fieldText: {
    fontSize: 20,
    lineHeight: 20,
    letterSpacing: 0.14,
  },
  selectIndicator: {
    width: 20,
    height: 20,
    marginRight: 8,
  },
  notSelected: {
    borderColor: Colors.black,
    opacity: 0.3,
  },
  notSelectedScale: {
    width: 20,
    height: 20,
    borderWidth: 1,
    borderRadius: 10,
  },
})
