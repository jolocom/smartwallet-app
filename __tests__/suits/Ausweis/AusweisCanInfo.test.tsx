import React from 'react'
import { useGoBack } from '~/hooks/navigation'
import { AusweisCanInfo } from '~/screens/LoggedIn/eID/components/AusweisCanInfo'
import { renderWithSafeArea } from '../../utils/renderWithSafeArea'

jest.mock('../../../src/hooks/navigation')

describe('Ausweis CAN info screen', () => {
  beforeAll(() => {
    ;(useGoBack as jest.Mock).mockReturnValue(jest.fn())
  })
  afterAll(() => {
    jest.resetAllMocks()
  })
  test('is displayed according to the designs', () => {
    const { toJSON } = renderWithSafeArea(<AusweisCanInfo />)
    expect(toJSON()).toMatchSnapshot()
  })
})
