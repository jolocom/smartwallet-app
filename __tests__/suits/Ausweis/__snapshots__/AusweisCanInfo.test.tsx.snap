// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ausweis CAN info screen is displayed according to the designs 1`] = `
<RNCSafeAreaView
  onInsetsChange={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        },
        Object {
          "flex": 1,
          "paddingTop": 0,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-start",
            "flex": 1,
            "justifyContent": "center",
          },
          false,
        ]
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#101010",
            "height": 0,
            "width": "100%",
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 50,
              "paddingHorizontal": 12,
              "width": "100%",
            },
            Object {
              "justifyContent": "space-between",
            },
            Array [
              Object {
                "backgroundColor": "#101010",
              },
              Object {},
            ],
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "height": 40,
              "justifyContent": "center",
              "opacity": 1,
              "width": 40,
            }
          }
          testID="backArrow"
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={17}
            bbWidth={25}
            focusable={false}
            height={17}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "flex": 0,
                  "height": 17,
                  "width": 25,
                },
              ]
            }
            vbHeight={17}
            vbWidth={25}
            width={25}
          >
            <RNSVGGroup>
              <RNSVGPath
                d="M24.48 7.438H3.34l5.362-5.469a.536.536 0 00.152-.375V.53A.53.53 0 008.334 0a.516.516 0 00-.369.156L.153 8.124a.539.539 0 000 .752l7.812 7.968c.15.152.372.198.568.116a.53.53 0 00.321-.491v-1.063a.536.536 0 00-.152-.375L3.34 9.562h21.138c.288 0 .521-.237.521-.53V7.968a.526.526 0 00-.52-.531z"
                fill={-855310}
                fillRule={1}
                propList={
                  Array [
                    "fill",
                    "fillRule",
                  ]
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "paddingHorizontal": 10,
            }
          }
          testID="heading"
        />
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
              "paddingHorizontal": 16,
              "width": "100%",
            },
            Object {
              "backgroundColor": "#101010",
              "marginBottom": -0,
              "paddingBottom": 0,
            },
            Object {},
            Object {
              "justifyContent": "flex-start",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "paddingTop": undefined,
                "textAlign": "center",
              },
              Object {
                "color": "rgba(255,255,255,0.85)",
                "fontFamily": "TTCommons-Medium",
                "fontSize": 28,
                "letterSpacing": 0,
                "lineHeight": 32,
              },
              undefined,
            ]
          }
        >
          AusweisCanInfo.header
        </Text>
        <Text
          style={
            Array [
              Object {
                "paddingTop": undefined,
                "textAlign": "center",
              },
              Object {
                "color": "rgba(255, 255, 255, 0.7)",
                "fontFamily": "TTCommons-Regular",
                "fontSize": 20,
                "letterSpacing": 0.14,
                "lineHeight": 22,
              },
              Object {
                "marginTop": 24,
                "textAlign": "left",
              },
            ]
          }
        >
          AusweisCanInfo.description
        </Text>
        <View
          style={
            Object {
              "aspectRatio": 1.7635467980295567,
              "marginTop": 36,
              "width": "100%",
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight="100%"
            bbWidth="100%"
            focusable={false}
            height="100%"
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "flex": 0,
                  "height": "100%",
                  "width": "100%",
                },
              ]
            }
            vbHeight={203}
            vbWidth={358}
            width="100%"
          >
            <RNSVGGroup>
              <RNSVGDefs>
                <RNSVGRect
                  height={203}
                  name="prefix__a"
                  rx={5.026}
                  width={358}
                  x={0}
                  y={0}
                />
              </RNSVGDefs>
              <RNSVGGroup
                fill={null}
                fillRule={0}
                propList={
                  Array [
                    "fill",
                    "fillRule",
                  ]
                }
              >
                <RNSVGMask
                  fill={-1}
                  height="100%"
                  maskContentUnits={1}
                  maskTransform={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      0,
                      0,
                    ]
                  }
                  maskUnits={0}
                  name="prefix__b"
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                  width="100%"
                  x="0%"
                  y="0%"
                >
                  <RNSVGUse
                    height={0}
                    href="prefix__a"
                    width={0}
                    x={0}
                    y={0}
                  />
                </RNSVGMask>
                <RNSVGRect
                  height={202.5}
                  propList={
                    Array [
                      "stroke",
                      "strokeWidth",
                    ]
                  }
                  rx={5.026}
                  stroke={-1}
                  strokeWidth={0.5}
                  width={357.5}
                  x={0.25}
                  y={0.25}
                />
                <RNSVGText
                  content={null}
                  dx={Array []}
                  dy={Array []}
                  fill={-6710887}
                  font={
                    Object {
                      "fontFamily": "TTCommons-Medium",
                      "fontSize": 13.922,
                      "fontWeight": 400,
                      "letterSpacing": 0.994,
                    }
                  }
                  mask="prefix__b"
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                  rotate={Array []}
                  x={Array []}
                  y={Array []}
                >
                  <RNSVGTSpan
                    content="PERSONALAUSWEIS"
                    dx={Array []}
                    dy={Array []}
                    font={Object {}}
                    rotate={Array []}
                    x={
                      Array [
                        15.906,
                      ]
                    }
                    y={
                      Array [
                        24.931,
                      ]
                    }
                  />
                </RNSVGText>
                <RNSVGRect
                  fill={-11645362}
                  height={25.076}
                  mask="prefix__b"
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                  rx={2.387}
                  width={107.4}
                  x={230.547}
                  y={135.029}
                />
                <RNSVGText
                  content={null}
                  dx={Array []}
                  dy={Array []}
                  fill={-1}
                  font={
                    Object {
                      "fontFamily": "TTCommons-Regular",
                      "fontSize": 20.048,
                      "letterSpacing": 2.864,
                    }
                  }
                  mask="prefix__b"
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                  rotate={Array []}
                  x={Array []}
                  y={Array []}
                >
                  <RNSVGTSpan
                    content="XXXXXX"
                    dx={Array []}
                    dy={Array []}
                    font={Object {}}
                    rotate={Array []}
                    x={
                      Array [
                        244.977,
                      ]
                    }
                    y={
                      Array [
                        153.812,
                      ]
                    }
                  />
                </RNSVGText>
                <RNSVGPath
                  d="M164.317 200.705c-8.34-8.346-14.901-13.199-19.681-14.557-13.459-3.825-22.029-14.163-22.029-24.548v-7.553c13.658-13.69 22.47-37.294 22.47-58.537 0-31.157-18.064-50.04-47.142-50.04-29.077 0-46.7 18.883-46.7 50.04 0 21.243 8.811 44.847 22.469 58.537v7.553c0 10.385-8.37 19.827-22.028 24.548-4.732 1.721-11.764 6.644-21.095 14.768l133.736-.21z"
                  fill={-11645362}
                  fillRule={1}
                  mask="prefix__b"
                  propList={
                    Array [
                      "fill",
                      "fillRule",
                    ]
                  }
                />
              </RNSVGGroup>
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
